transform/transformer.go

```diff
@@ -27,19 +38,20 @@ type Transformer struct {
 	schema              *jsonschema.Schema
 	transformIdentifier string // Used to select the proper transform Instructions
 	root                instanceTransformer
+	format              inputFormat
 }
 
 // NewTransformer returns a Transformer using the schema given.
 // The transformIdentifier is used to select the appropriate transform section from the schema.
-func NewTransformer(schema *jsonschema.Schema, tranformIdentifier string) (*Transformer, error) {
-	tr := &Transformer{schema: schema, transformIdentifier: tranformIdentifier}
-
+// The format is used to execute the correct transforms ('JSON' or 'XML')
+func NewTransformer(schema *jsonschema.Schema, tranformIdentifier string, format inputFormat) (*Transformer, error) {
```

Comment:
This is an exported function and we don't want to do any breaking changes. Basically keep this as is but assume the format is JSON and add a NewXMLTransformer which sets the format to xml. That avoids it being a breaking change.

Reply:
oh makes sense, incoming commit